File: backend/RoomBooking.API/Models/Booking.cs
namespace RoomBooking.API.Models;

public class Booking
{
    public int Id { get; set; }

    public int UserId { get; set; }

    public User? User { get; set; }

    public int DepartmentId { get; set; }

    public Department? Department { get; set; }

    public int RoomId { get; set; }

    public Room? Room { get; set; }

    public string Title { get; set; } = string.Empty;

    public string? Description { get; set; }

    public DateTime StartTime { get; set; }

    public DateTime EndTime { get; set; }

    public BookingStatus Status { get; set; }

    public string? RejectionReason { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime? UpdatedAt { get; set; }
}

File: frontend/room-booking-web/types/booking.ts
export type BookingStatus =
  | "Pending"
  | "Approved"
  | "Rejected"
  | "Cancelled";

export type Booking = {
  id: number;
  title: string;
  description?: string;
  roomId: number;
  roomName: string;
  userId: number;
  userName: string;
  startTime: string;
  endTime: string;
  status: BookingStatus;
};

export type CreateBookingRequest = {
  roomId: number;
  title: string;
  description?: string;
  startTime: string;
  endTime: string;
};